model/messagedeletions.go

```diff
@@ -0,0 +1,31 @@
+package model
+
+// MessageDeletions holds information about deleted messages.
+type MessageDeletions struct {
+	Messages []*Message
+}
+
+// ToExternal converts the event into an external representation.
+func (del MessageDeletions) ToExternal() interface{} {
+	res := &MessageDeletionsExternal{
+		make([]uint, len(del.Messages)),
+	}
+	for key, value := range del.Messages {
+		res.IDs[key] = value.ID
+	}
+	return res
+}
+
+// MessageDeletionsExternal Model
+//
+// MessageDeletionsExternal holds information about deletions which will be sent to the clients.
+//
+// swagger:model MessageDeletions
+type MessageDeletionsExternal struct {
```

Comment:
I've though a little about this. In case of a delete all we could send a "refresh state" event. Which just invalides all data on clientsides and forces a refetch. 

This is maybe easier to handle on clientside, as we don't have to check every message if it should be deleted.

What do you think?

Reply:
I'd argue that we would have a three-way message exchange then. The client needs to request all unread messages, and the server has to respond again. Doesn't that make things more power-consuming? Maybe just slightly...

What do you mean by checking every message on the client side? I don't think I understand what should be the challenge here.

Reply:
If the client has currently 100k messages in memory and all messages were deleted. Then the client would receive a message_deleted event with 100k messages. The client now has to remove every message with an id that is included inside the websocket event.

Anyway, I think the current solution is okay. The client can still decide what todo with the delete event. maybe just refreshes all its data when the websocket event contains more than X messages.